x-pack/plugins/security_solution/public/management/pages/policy/view/tabs/policy_tabs.tsx

```diff
@@ -131,12 +133,35 @@ export const PolicyTabs = React.memo(() => {
   );
   const isEnterprise = useLicense().isEnterprise();
   const isProtectionUpdatesEnabled = isEnterprise && isProtectionUpdatesFeatureEnabled;
+
+  const getHostIsolationExceptionsApiClientInstance = useCallback(
+    () => HostIsolationExceptionsApiClient.getInstance(http),
+    [http]
+  );
+
+  const hasAccessToHostIsolationExceptions = useHostIsolationExceptionsAccess(
+    canAccessHostIsolationExceptions,
+    canReadHostIsolationExceptions,
+    getHostIsolationExceptionsApiClientInstance
+  );
+
+  const hostIsolationExceptionsAccessLoading = hasAccessToHostIsolationExceptions === null;
+
   // move the user out of this route if they can't access it
   useEffect(() => {
+    if (hostIsolationExceptionsAccessLoading) {
+      return;
+    }
+
+    const redirectHostIsolationException =
+      isInHostIsolationExceptionsTab &&
+      (!canReadHostIsolationExceptions ||
+        (!hostIsolationExceptionsAccessLoading && !hasAccessToHostIsolationExceptions));
```

Comment:
Hm, this code block in itself is used only for redirects if user lands on a page they have no access to. We cannot determine whether it's true if any of the two resources is still loading, therefore, if loading, then don't execute it at all. From the UI/UX perspective, if any of these two is true we display a loader in a render return. 
If no loader was present it could happen that user lands on a rendered page just to be redirected once resources are fetched.